src/main/proto/netflix/titus/titus_job_api.proto

```diff
@@ -83,6 +83,22 @@ message Image {
   string digest = 3;
 }
 
+// Definition of a request to add a platform sidecar alongside a task
+// Note that this is *not* a user-defined sidecar, that is why it just has a
+// name. These platform-sidecars are attached a task start time, and the
+// definition of what the sidecar is is not baked into the job itself, just the
+// intent.
+message PlatformSidecar {
+  // (Required) Name of the platform sidecar requested
+  string name = 1;
+
+  // (Optional) Channel representing a pointer to releases of the sidecar
+  string channel = 2;
+
+  // (Optional) Arguments, KV pairs for configuring the sidecar
+  map<string, string> constraints = 3;
```

Comment:
I don't think these are "constraints" so much as "parameters" or "arguments", since these KV pairs are expected to be validated against schemas stored external to this API. Which makes me think this field should really be _dynamically typed_ rather than explicitly `map<string,string>`. I'm not sure how practical that is for the Titus Job API, but it seems reasonable for it to be [opaque at the protobuf level](https://blog.envoyproxy.io/dynamic-extensibility-and-protocol-buffers-dcd0bf0b8801).

Reply:
+1. This term is used in the job descriptor to configure placement constraints. To pass arbitrary data we have `attributes` fields in other messages.

Reply:
Whoops! This was a mistake, this meant to be arguments